src/components/PostDetails.tsx

```diff
@@ -2,106 +1,93 @@
+  selectedPost: selectedPost,
+}) => {
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [isCommentsLoading, setIsCommentsLoading] = useState(false);
+  const [hasError, setHasError] = useState(false);
+  const [isCommentFormShown, setIsCommentFormShown] = useState(false);
 
-          <p className="title is-4">Comments:</p>
+  useEffect(() => {
+    setIsCommentsLoading(true);
+    commentService
+      .getComments(selectedPost.id)
+      .then(setComments)
+      .catch(() => setHasError(true))
+      .finally(() => setIsCommentsLoading(false));
+  }, [selectedPost]);
 
-          <article className="message is-small" data-cy="Comment">
-            <div className="message-header">
-              <a href="mailto:misha@mate.academy" data-cy="CommentAuthor">
-                Misha Hrynko
-              </a>
-              <button
-                data-cy="CommentDelete"
-                type="button"
-                className="delete is-small"
-                aria-label="delete"
-              >
-                delete button
-              </button>
-            </div>
+  function deleteComment(commId: number) {
+    setComments(currComments => currComments.filter(com => commId !== com.id));
 
-            <div className="message-body" data-cy="CommentBody">
-              Some comment
-            </div>
-          </article>
+    return commentService.deleteComment(commId).catch(error => {
+      setComments(comments);
+      setHasError(true);
+      throw error;
+    });
+  }
 
-          <article className="message is-small" data-cy="Comment">
-            <div className="message-header">
-              <a href="mailto:misha@mate.academy" data-cy="CommentAuthor">
-                Misha Hrynko
-              </a>
+  return (
+    <div className="content" data-cy="PostDetails">
+      <div className="block">
+        <h2 data-cy="PostTitle">
+          #{selectedPost.id}: {selectedPost.title}
+        </h2>
 
-              <button
-                data-cy="CommentDelete"
-                type="button"
-                className="delete is-small"
-                aria-label="delete"
-              >
-                delete button
-              </button>
-            </div>
-            <div className="message-body" data-cy="CommentBody">
-              One more comment
-            </div>
-          </article>
+        <p data-cy="PostBody">{selectedPost.body}</p>
+      </div>
 
-          <article className="message is-small" data-cy="Comment">
-            <div className="message-header">
-              <a href="mailto:misha@mate.academy" data-cy="CommentAuthor">
-                Misha Hrynko
-              </a>
+      <div className="block">
+        {isCommentsLoading ? (
+          <Loader />
+        ) : (
+          <>
+            {hasError && (
+              <div className="notification is-danger" data-cy="CommentsError">
+                Something went wrong
+              </div>
+            )}
```

Comment:
Ensure that the error message provides actionable feedback to the user. Consider adding a retry mechanism or more detailed error information.